src/applications/ivc-champva/10-10D/helpers/utilities.js

```diff
@@ -103,12 +91,20 @@ export function populateFirstApplicant(
 // Only show address dropdown if we're not the certifier
 // AND there's another address present to choose from:
 export function page15aDepends(formData, index) {
-  const certifierIsApp =
-    get('certifierRelationship.relationshipToVeteran.applicant', formData) ===
-    true;
+  const certifierIsApp = get('certifierRole', formData) === 'applicant';
   const certAddress = get('street', formData?.certifierAddress);
 
   return (
     (index && index > 0) || (certAddress && !(certifierIsApp && index === 0))
   );
 }
+
+// Compare two objects (with the same keys) and identify differences
+export function objDiff(obj1, obj2, ignoreViewOnlyProps = true) {
+  return Object.keys(obj1).filter(
+    k =>
+      (obj1[k] && obj2[k]) !== undefined &&
+      obj1[k] !== obj2[k] &&
+      !(ignoreViewOnlyProps && k.includes('view:')),
+  );
+}
```

Comment:
Helper function - this gets used when we run the validation checks to make sure that every field of the signer's address is the same as every field of the sponsor (e.g., if the signer is the sponsor, they need to match field-for-field.)